### PR TITLE
Add Jscrambler Code Integrity starter workflow

### PR DESCRIPTION
**Pull request by vitorveiga**
## Pre-requisites

- [x] Prior to submitting a new workflow, please apply to join the GitHub Technology Partner Program: [partner.github.com/apply](https://partner.github.com/apply?partnershipType=Technology+Partner).

---

### **Please note that at this time we are only accepting new starter workflows for Code Scanning. Updates to existing starter workflows are fine.**

---

## Tasks

**For _all_ workflows, the workflow:**

- [x] Should be contained in a `.yml` file with the language or platform as its filename, in lower, [_kebab-cased_](https://en.wikipedia.org/wiki/Kebab_case) format (for example, [`docker-image.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-image.yml)).  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").
- [x] Should use sentence case for the names of workflows and steps (for example, "Run tests").
- [x] Should be named _only_ by the name of the language or platform (for example, "Go", not "Go CI" or "Go Build").
- [x] Should include comments in the workflow for any parts that are not obvious or could use clarification.
- [x] Should specify least privileged [permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token) for `GITHUB_TOKEN` so that the workflow runs successfully.

**For _CI_ workflows, the workflow:**

- [x] Should be preserved under [the `ci` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [x] Should include a matching `ci/properties/*.properties.json` file (for example, [`ci/properties/docker-publish.properties.json`](https://github.com/actions/starter-workflows/blob/main/ci/properties/docker-publish.properties.json)).
- [x] Should run on `push` to `branches: [ $default-branch ]` and `pull_request` to `branches: [ $default-branch ]`.
- [x] Packaging workflows should run on `release` with `types: [ created ]`.
- [x] Publishing workflows should have a filename that is the name of the language or platform, in lower case, followed by "-publish" (for example, [`docker-publish.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-publish.yml)).

**For _Code Scanning_ workflows, the workflow:**

- [x] Should be preserved under [the `code-scanning` directory](https://github.com/actions/starter-workflows/tree/main/code-scanning).
- [x] Should include a matching `code-scanning/properties/*.properties.json` file (for example, [`code-scanning/properties/codeql.properties.json`](https://github.com/actions/starter-workflows/blob/main/code-scanning/properties/codeql.properties.json)), with properties set as follows:
  - [x] `name`: Name of the Code Scanning integration.
  - [x] `creator`: Name of the organization/user producing the Code Scanning integration.
  - [x] `description`: Short description of the Code Scanning integration.
  - [x] `categories`: Array of languages supported by the Code Scanning integration.
  - [x] `iconName`: Name of the SVG logo representing the Code Scanning integration. This SVG logo must be present in [the `icons` directory](https://github.com/actions/starter-workflows/tree/main/icons).
- [ ] Should run on `push` to `branches: [ $default-branch, $protected-branches ]` and `pull_request` to `branches: [ $default-branch ]`. We also recommend a `schedule` trigger of `cron: $cron-weekly` (for example, [`codeql.yml`](https://github.com/actions/starter-workflows/blob/c59b62dee0eae1f9f368b7011cf05c2fc42cf084/code-scanning/codeql.yml#L14-L21)).

**Some general notes:**

- [x] This workflow must _only_ use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
- [x] This workflow must _only_ use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  We require that these actions be referenced using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [x] Automation and CI workflows should not send data to any 3rd party service except for the purposes of installing dependencies.
- [x] Automation and CI workflows cannot be dependent on a paid service or product.

### CODE REVIEW

File: ci/jscrambler-code-integrity.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+# This is a basic workflow to help you get started with Using Jscrambler Code Integrity Action,
+# documentation can be found here : https://docs.jscrambler.com/latest/code-integrity/documentation/github-ci-integration
+
+name: Jscrambler Code Integrity
+
+on:
+  release:
+    types: [created]
+
```
Comment (Phantsure) [inline ci/jscrambler-code-integrity.yml:14]: Can we have workflow level permissions?

File: ci/jscrambler-code-integrity.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (Phantsure) [inline ci/jscrambler-code-integrity.yml:17]: Job permission would also be recommended. Please add those

File: ci/jscrambler-code-integrity.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+# This is a basic workflow to help you get started with Using Jscrambler Code Integrity Action,
```
Comment (Phantsure) [inline ci/jscrambler-code-integrity.yml:6]: Can you add what does this worklow/action do, how it does that and what all is required for it to run?

File: ci/properties/jscrambler-code-integrity.properties.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+    "name": "Jscrambler Code Integrity",
+    "description": "Protect your JavaScript Application with polymorphic obfuscation, code locks, and self-defensive techniques",
+    "creator": "Jscrambler",
+    "iconName": "jscrambler",
+    "categories": ["Continuous integration", "JavaScript"]
```
Comment (Phantsure) [inline ci/properties/jscrambler-code-integrity.properties.json:6]: Does code-scanning feel like a better category?

Comment (vitorveiga) [reply ci/properties/jscrambler-code-integrity.properties.json:6]: Not really! The Code Integrity Github action applies some transformations to the user's source code. Thus calling it Code Scanning can be misleading.

File: ci/jscrambler-code-integrity.yml (hunk 1 above)
Comment (vitorveiga) [reply ci/jscrambler-code-integrity.yml:14]: Sure

File: ci/jscrambler-code-integrity.yml (hunk 3 above)
Comment (vitorveiga) [reply ci/jscrambler-code-integrity.yml]: Sure

File: ci/properties/jscrambler-code-integrity.properties.json (hunk 4 above)
Comment (vitorveiga) [reply ci/properties/jscrambler-code-integrity.properties.json:6]: Security is also a good candidate for this starter workflow. Any feedback on this matter?

Comment (Phantsure) [reply ci/properties/jscrambler-code-integrity.properties.json:6]: We currently keep security templates like code-scanning and dependency review in `code-scanning` directory. Does "Code Scanning" category fit well under security template?

Comment (liamgutierrez77) [reply ci/properties/jscrambler-code-integrity.properties.json:6]: Are you appreciate? Is it everything you guys are doing a no for all my heart you're doing the great work and great day and I appreciate it so much that you guys are helping me a lot. Also wanted to give my information iPhone that I have if I can send the pictures or video. I think they have my break jail for Miami. I don't know but I continue hacking my phone it is it's crazy.imageimageimage

File: ci/jscrambler-code-integrity.yml (hunk 1 above)
Comment (liamgutierrez77) [reply ci/jscrambler-code-integrity.yml:17]: Pha why you close the issue never finished  never asking you to close whit consent

File: ci/properties/jscrambler-code-integrity.properties.json (hunk 4 above)
Comment (vitorveiga) [reply ci/properties/jscrambler-code-integrity.properties.json]: Moved to `code-scanning`. Thanks

**Review by Phantsure — APPROVED** [review]

**Comment by Phantsure** [comment]
@vitorveiga Can you rebase this and ensure no checks are failing?

**Comment by vitorveiga** [comment]
> @vitorveiga Can you rebase this and ensure no checks are failing?

Reviewed and Updated!

**Comment by griffinashe** [comment]
@Phantsure @vitorveiga - I think we need to move this out of the Code Scanning section. I see the previous discussion above about where this should live. Since this isn't a code scanning technology, that category isn't the best place for it.

**Pull request merged**